### PR TITLE
🚀 Transformer for removing AMP devAssert, working on existing known cases

### PR DESCRIPTION
**Pull request by kristoferbaxter**
This WIP plugin is ready for review, marked as a draft since it's not yet tied into the development process.

### CODE REVIEW

File: build-system/babel-plugins/babel-plugin-amp-dev-assert-transformer/index.js [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Is a MemberExpression "thenable", `foo().then(() => {});`
```
Comment (kevinkimball) [inline build-system/babel-plugins/babel-plugin-amp-dev-assert-transformer/index.js:18]: Isn't this just asking if the expression is a promise or not? What else would be "thenable" besides a promise?

File: build-system/babel-plugins/babel-plugin-amp-dev-assert-transformer/index.js [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+   * @return {boolean}
+   */
+  function handleInnerMemberExpression(callExpression, memberExpression) {
+    const innerCallExpression = getMatchingInnerCallExpression(
+      memberExpression
+    );
+    if (innerCallExpression) {
+      const evalutedDevAssert = evaluateDevAssert(innerCallExpression);
+      if (!evalutedDevAssert.confident) {
+        return;
```
Comment (erwinmombay) [inline build-system/babel-plugins/babel-plugin-amp-dev-assert-transformer/index.js:104]: should we return a boolean here?

**Comment by rsimha** [comment]
Drive-by comment to say thanks for working on this PR, since it will have a cascading effect of improving several aspects of our compilation toolchain. Some context:

- This PR fixes #23960 (_Not all assert calls are being removed from v0.js by babel transforms_)
- Fixing #23960 will unblock #24047 (_Remove `AmpPass.java`_)
- Merging #24047 will eliminate the `runner.jar` concurrency problems described in https://github.com/ampproject/amphtml/issues/24487#issuecomment-621471530 (_`gulp dist` is frequently timing out on Travis_)
- This should allow for significantly faster compilation on high-end machines with lots of cores

**Comment by kristoferbaxter** [comment]
Likely won’t be able to work on this for some time, but this PR might be a good jumping off point for someone to complete the work.

**Pull request closed without merging**